### PR TITLE
python2Packages.pyezminc: init at 1.2

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [NA] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [NA] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [NA] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/pyezminc/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ buildPythonPackage, isPy3k, fetchFromGitHub, stdenv,
+  netcdf, hdf5, libminc, ezminc,
+  cython, numpy, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "pyezminc";
+  version = "1.2";
+  name = "${pname}-${version}";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyezminc/default.nix]: drop this line

File: pkgs/development/python-modules/pyezminc/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ 
+  disabled = isPy3k;  # ??
+  # sort of works but the example programs fail due to unicode issues, etc.
+
+  src = fetchFromGitHub {
+    owner  = "BIC-MNI";
+    repo   = "pyezminc";
+    rev    = "b9fc331c5ad296348fa1ac39df8b0e26ed667d85";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyezminc/default.nix]: `rev = "release-${version}"`

File: pkgs/development/python-modules/pyezminc/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{ buildPythonPackage, isPy3k, fetchFromGitHub, stdenv,
+  netcdf, hdf5, libminc, ezminc,
+  cython, numpy, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "pyezminc";
+  version = "1.2";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyezminc/default.nix]: `1.2.01`

File: pkgs/development/python-modules/pyezminc/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    repo   = "pyezminc";
+    rev    = "b9fc331c5ad296348fa1ac39df8b0e26ed667d85";
+    sha256 = "1qf115d82pyw956r9jcl9xxy03smfxz7ayj1chixd1yaxsqvcigs";
+  };
+
+  nativeBuildInputs = [ cython ];
+  buildInputs = [ netcdf hdf5 libminc ezminc ];
+  propagatedBuildInputs = [ numpy scipy ];
+
+  preBuild = "export CFLAGS='-fpermissive'";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyezminc/default.nix]: I think there's some more standard way to do this. Please grep through Nixpkgs.

File: pkgs/development/python-modules/pyezminc/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  netcdf, hdf5, libminc, ezminc,
+  cython, numpy, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "pyezminc";
+  version = "1.2";
+  name = "${pname}-${version}";
+ 
+  disabled = isPy3k;  # ??
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyezminc/default.nix]: README says

> BUILD REQUIREMENTS:
> - python2-devel

so I think it should remain disabled for Python 3. Please update the comment.

File: pkgs/development/python-modules/pyezminc/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  preBuild = "export CFLAGS='-fpermissive'";
+
+  doCheck = false;  # e.g., expects test data in /opt
+
+  meta = {
+    homepage = https://github.com/BIC-MNI/pyezminc;
+    description = "Python API for libminc using EZMINC";
+    license = stdenv.lib.licenses.gpl2;
+    maintainers = with stdenv.lib.maintainers; [ bcdarwin ];
+    platforms = stdenv.lib.platforms.unix;
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyezminc/default.nix]: drop this line

File: pkgs/development/python-modules/pyezminc/default.nix [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 = "1qf115d82pyw956r9jcl9xxy03smfxz7ayj1chixd1yaxsqvcigs";
+  };
+
+  nativeBuildInputs = [ cython ];
+  buildInputs = [ netcdf hdf5 libminc ezminc ];
+  propagatedBuildInputs = [ numpy scipy ];
+
+  preBuild = "export CFLAGS='-fpermissive'";
+
+  doCheck = false;  # e.g., expects test data in /opt
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyezminc/default.nix]: Probably needs patching in `postPatch`: https://github.com/BIC-MNI/pyezminc/search?q=opt&unscoped_q=opt

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pyezminc/default.nix (hunk 7 above)
Comment (dotlambda) [reply pkgs/development/python-modules/pyezminc/default.nix]: Is there a package that includes `share/icbm152_model_09c`?

File: pkgs/development/python-modules/pyezminc/default.nix [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  cython, numpy, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "pyezminc";
+  version = "1.2";
+  name = "${pname}-${version}";
+ 
+  disabled = isPy3k;  # ??
+  # sort of works but the example programs fail due to unicode issues, etc.
```
Comment (Assassinkin) [inline pkgs/development/python-modules/pyezminc/default.nix]: Did you try setting these:
`LC_ALL = "en_US.utf-8";`
`nativeBuildInputs = [ glibcLocales ];`

File: pkgs/development/python-modules/pyezminc/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pyezminc/default.nix]: I don't understand this comment

File: pkgs/development/python-modules/pyezminc/default.nix (hunk 2 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pyezminc/default.nix]: done

File: pkgs/development/python-modules/pyezminc/default.nix (hunk 3 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pyezminc/default.nix]: done

File: pkgs/development/python-modules/pyezminc/default.nix (hunk 4 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pyezminc/default.nix]: changed to NIX_CFLAGS_COMPILE

File: pkgs/development/python-modules/pyezminc/default.nix (hunk 6 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pyezminc/default.nix]: I don't understand this comment

Comment (dotlambda) [reply pkgs/development/python-modules/pyezminc/default.nix]: `platforms` is set automatically by `buildPythonPackage`

File: pkgs/development/python-modules/pyezminc/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/pyezminc/default.nix]: `name` is set automatically by `buildPythonPackage`

File: pkgs/development/python-modules/pyezminc/default.nix (hunk 5 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pyezminc/default.nix]: comment removed

File: pkgs/development/python-modules/pyezminc/default.nix (hunk 7 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pyezminc/default.nix]: I think they're available as a tarball somewhere so in principle they could be included in this package or another one (e.g. BEAST, not yet Nixified)

File: pkgs/development/python-modules/pyezminc/default.nix (hunk 6 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pyezminc/default.nix]: removed

File: pkgs/development/python-modules/pyezminc/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pyezminc/default.nix]: removed

**Comment by FRidh** [comment]
@GrahamcOfBorg build python2Packages.pyezminc python3Packages.pyezminc

**Pull request merged**